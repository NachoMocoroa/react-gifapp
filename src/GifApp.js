import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="app">
            <header>
                <h1>GifApp</h1>
                <AddCategory setCategories={ setCategories } />
            </header>
            <main>
                {
                    categories.map(category => (
                        <GifGrid 
                            key={ category } 
                            category={ category } 
                        />
                    ))
                }
            </main>
            <footer>
                <p>All images from &copy;GIPHY, Inc.</p>
            </footer>
        </div>
    );
        
}

export default GifApp;