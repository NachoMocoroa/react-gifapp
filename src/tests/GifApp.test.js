import React from 'react';
import { shallow } from 'enzyme';
import GifApp from '../GifApp';

describe('Test GifApp', () => {

    test('should render GifApp component', () => {
        const wrapper = shallow(<GifApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show categories list', () => {
        const categories = ['gif'];
        const wrapper = shallow(<GifApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});
