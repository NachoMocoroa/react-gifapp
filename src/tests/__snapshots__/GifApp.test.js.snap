// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifApp should render GifApp component 1`] = `
<div
  className="app"
>
  <header>
    <h1>
      GifApp
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </header>
  <main />
  <footer>
    <p>
      All images from ©GIPHY, Inc.
    </p>
  </footer>
</div>
`;

exports[`Test GifApp should show categories list 1`] = `
<div
  className="app"
>
  <header>
    <h1>
      GifApp
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </header>
  <main>
    <GifGrid
      category="gif"
      key="gif"
    />
  </main>
  <footer>
    <p>
      All images from ©GIPHY, Inc.
    </p>
  </footer>
</div>
`;
